src/components/map/MapService.js

```diff
@@ -828,14 +804,42 @@ goog.require('ga_urlutils_service');
           }
         };
 
-        /**
-         * Reurn an array of pre-selected bodId from current topic
-         */
-        this.getSelectedLayers = function() {
-          if (!layers || !gaTopic.get()) {
+        // Load layers config
+        var lastLangUsed;
+        var loadLayersConfig = function(lang) {
+          if (lastLangUsed == lang) {
             return;
           }
-          return gaTopic.get().selectedLayers;
+          lastLangUsed = lang;
+          var url = getLayersConfigUrl(lang);
+          return $http.get(url).then(function(response) {
```

Comment:
I think we need to use the same machanism in gaTopic as is used here. Here, we publish the promise returned by the `then` function. In gaTopic, publish the promise returned by the request.